src/Symfony/Component/Intl/Data/Generator/LocaleDataGenerator.php

```diff
@@ -158,22 +142,26 @@ protected function generateDataForMeta(BundleEntryReaderInterface $reader, $temp
     /**
      * @return string
      */
-    private function generateLocaleName($locale, $displayLocale, $pattern, $separator)
+    private function generateLocaleName(BundleEntryReaderInterface $reader, $tempDir, $locale, $displayLocale, $pattern, $separator)
     {
         // Apply generic notation using square brackets as described per http://cldr.unicode.org/translation/language-names
-        $name = str_replace(['(', ')'], ['[', ']'], $this->languageDataProvider->getName(\Locale::getPrimaryLanguage($locale), $displayLocale));
+        $name = str_replace(['(', ')'], ['[', ']'], $reader->readEntry($tempDir.'/lang', $displayLocale, ['Languages', \Locale::getPrimaryLanguage($locale)]));
```

Comment:
now uses `IntlBundleReader` instead of JsonBundleReader` ... total compilation time;

```
3.4
real    1m40.170s

PR
real    0m22.073s
```

moreover, this allows to keep the generators decoupled